crates/iota-indexer/tests/rpc-tests/read_api.rs

```diff
@@ -1234,18 +1239,83 @@ fn try_get_past_object() {
         runtime,
         store,
         client,
-        ..
+        cluster,
     } = ApiTestSetup::get_or_init();
+
     runtime.block_on(async move {
         indexer_wait_for_checkpoint(store, 1).await;
 
+        let object_id = ObjectID::random();
+        let version = SequenceNumber::new();
+
+        // Test: Object does not exist
         let result = client
-            .try_get_past_object(ObjectID::random(), SequenceNumber::new(), None)
+            .try_get_past_object(object_id, version, None)
+            .await
+            .expect("RPC call should succeed");
+
+        assert!(
+            matches!(result, IotaPastObjectResponse::ObjectNotExists(_object_id)),
+            "Expected NotExists response, got: {:?}",
+            result
+        );
+
+        // Create a valid object
+        let (sender, _): (_, AccountKeyPair) = get_key_pair();
+
+        let gas = cluster
+            .fund_address_and_return_gas(
+                cluster.get_reference_gas_price().await,
+                Some(10_000_000_000),
+                sender,
+            )
             .await;
-        assert!(rpc_call_error_msg_matches(
-            result,
-            r#"{"code":-32601,"message":"Method not found"}"#
-        ));
+
+        indexer_wait_for_object(client, gas.0, gas.1).await;
+
+        // Test: Version Found
+        let result = client
+            .try_get_past_object(gas.0, gas.1, None)
+            .await
+            .expect("RPC call should succeed");
+
+        assert!(
+            matches!(result, IotaPastObjectResponse::VersionFound { .. }),
```

Comment:
As above, if we know exact expected value I think it would be better to use `assert_eq`. (I think it produces nicer error messages?)